### PR TITLE
Draft PR Zonelimit

### PR DESCRIPTION
**Pull request by TCY16**

### CODE REVIEW

**Comment by TCY16** [comment]
I've made a pull request so we can review this branch easier. I noticed there are changes to the Cookies files in here as well, we should discuss if we want these here.

File: ZoneLimit/xdp_zonelimit.c [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+	struct cursor     c;
+	struct meta_data *md = (void *)(long)ctx->data_meta;
+
+	cursor_init(&c, ctx);
+	if ((void *)(md + 1) > c.pos)
+		return XDP_ABORTED;
+
+    struct ethhdr *eth;
+    struct udphdr *udp;
+	struct dnshdr    *dns;
```
Comment (TCY16) [inline ZoneLimit/xdp_zonelimit.c:174]: formatting
```suggestion
    struct cursor     c;
    struct meta_data *md = (void *)(long)ctx->data_meta;

    cursor_init(&c, ctx);
    if ((void *)(md + 1) > c.pos)
        return XDP_ABORTED;

    struct ethhdr *eth;
    struct udphdr *udp;
    struct dnshdr *dns;
```

File: ZoneLimit/xdp_zonelimit.c [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+struct bpf_map_def SEC("maps") jmp_table = {
+    .type = BPF_MAP_TYPE_PROG_ARRAY,
+    .key_size = sizeof(uint32_t),
+    .value_size = sizeof(uint32_t),
+    .max_entries = 4
+};
+
+#define HANDLE_MATCH 0
+#define CHECK_CACHE 1
+#define PARSE_DNAME 2
```
[the remark this answers is not in the document]
Comment (DRiKE) [reply ZoneLimit/xdp_zonelimit.c:134]: Those are used in the `bpf_tail_call`s

File: ZoneLimit/xdp_zonelimit.c [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+        if (c.pos + labellen + 1 > c.end) {
+            return XDP_PASS;
+        }
+        //res->len += labellen + 1;
+        //in_packet_len += labellen + 1;
+        c.pos += labellen + 1;
+    }
+    if (c.pos + 1 > c.end) {
+        return XDP_PASS;
+    }
```
[the remark this answers is not in the document]
Comment (DRiKE) [reply ZoneLimit/xdp_zonelimit.c:361]: Yes, as `c.pos` will have been updated in the for-loop.

**Comment by DRiKE** [comment]
Some 'please add a comment here'-comments are not addressed yet, I want to go over the entire thing again anyway after we wrote up the blog post for this.

Thanks!

**Pull request merged**